src/DataflowPass.cpp

```diff
@@ -168,79 +205,132 @@ void DataflowPass::transfer(Instruction *instruction,
       bool loopBroken = false;
       for (auto Pair : this->memoryFunctions) {
         if (fnName == Pair.first) {
-          logout("not calling onalloc for "
-                 << arg << " of fnName " << fnName << " and inst "
-                 << *instruction << " and just returning now") return;
+          logout("not calling onalloc for " << arg << " of fnName " << fnName
+                                            << " and inst " << *instruction
+                                            << " and just returning now");
+          return;
 
         } else if (fnName == Pair.second) {
-          MethodsSet *methods =
-              inputProgramPoint.getPVRef(arg, false)->getMethodsSetRef();
-          logout("GETMAIN-5 ON '"
-                 << arg
-                 << "'") this->onDeallocationFunctionCall(methods, fnName);
+          MethodsSet *methods = pv->getMethodsSetRef();
+          logout("GETMAIN-5 ON '" << arg << "'");
+          this->onDeallocationFunctionCall(methods, fnName);
           loopBroken = true;
           break;
         }
       }
 
-      if (loopBroken) {
+      if (loopBroken || fnName == "llvm.dbg.declare") {
         continue;
       }
 
-      if (fnName == "llvm.dbg.declare") {
-        continue;
+      // finds annotations
+      logout("finding annotations for " << fnName);
+
+      /*
+        LLVM uses sret attributes as an optimization if the return type is
+        large. a struct with more than 2 fields can trigger this optimization.
+
+        docs: https://llvm.org/docs/LangRef.html#parameter-attributes
+        less credible forum post:
+        https://discourse.llvm.org/t/optimizing-sret-on-caller-side/60660
+
+        our instructions end up looking like this:
+        call void @does_something_a(%struct.my_struct* sret %A1,
+        %struct.my_struct* byval align 8 %A), !dbg !58
+
+        here's how we will handle these types of instructions:
+
+        say we have the instruction
+        call void @does_something_a(%struct.my_struct* sret %A1,
+        %struct.my_struct* byval align 8 %A), !dbg !58
+
+        (this is from C code
+        my_struct A1 = does_something_a(A);)
```

Comment:
Please use better names than "A1" vs "A": this is hard enough to keep track of as it is :)

Maybe "A" and "B"?